app/controllers/herbaria/alls_controller.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Herbaria
+  # Controls viewing and modifying herbaria.
+  class AllsController < ApplicationController
+    # filters
+    before_action :store_location
+
+    # Old MO Action (method)        New "Normalized" Action (method)
+    # ----------------------        --------------------------------
+    # list_herbaria (get)           Herbaria::All#index (get)
+
+    # ---------- Actions to Display data (index, show, etc.) -------------------
+
+    # Display all Herbaria
+    # linked (conditionally) from HerbariaIndex
+    def index
+      query = create_query(:Herbarium, :all, by: :name)
+      show_selected_herbaria(query, always_index: true)
+    end
```

Comment:
Questions:
- **Should I combine all the index-type actions as `Herbaria#index`** by: 
  - always calling Herbaria#index with a parameter (e.g., `index-type`)
  - using a `case` statement that depends on that parameter.
- **Should I interchange `Herbaria#index` and `Herbaria::Alls#index`?**  
This action is the former `Herbarium#list`, and lists _all_ Herbaria. I think the action that lists all model instances would typically be `#index`. It turned out to be _much_ simpler to translate the old actions as follows:

| old | new |
| --- | ---|
| Herbarium#index | Herbaria::Nonpersonals#index |
| Herbarium#index_herbarium | Herbaria#index - lists query results |
| Herbarium#list_herbaria | Herbaria::Alls#index - all herbaria |

Reply:
My initial reaction is, "yes" to question 1.  I feel like just on principle, the "Alls" controller seems like it should be unnecessary.

But that seems easy to accomplish if we follow the pattern where Herbaria#index (or any controller's index action) would either list all herbaria, or list *selected* herbaria if it's passed a q param. 

fwiw this is something i had been experimenting with, but i haven't poked around enough to see if you've got an example of that available elsewhere here.